src/serializer/Referentializer.js

```diff
@@ -54,13 +53,17 @@ export class Referentializer {
 
   _options: SerializerOptions;
   scopeNameGenerator: NameGenerator;
-  statistics: SerializerStatistics;
   realm: Realm;
 
   _newCapturedScopeInstanceIdx: number;
   referentializationState: Map<ReferentializationScope, ReferentializationState>;
   _referentializedNameGenerator: NameGenerator;
 
+  getStatistics(): SerializerStatistics {
+    invariant(this.realm.statistics instanceof SerializerStatistics);
```

Comment:
These invariants make me feel uneasy. As always, I strongly recommend adding a comment explaining why this must always be true. I can't say that I understand the changes in this PR well enough to convince myself about this.